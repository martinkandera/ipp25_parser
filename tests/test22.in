class Main : Object {
    run [|
        x := Integer new.
    ]
}