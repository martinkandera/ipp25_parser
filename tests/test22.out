<?xml version="1.0" encoding="UTF-8"?>
<program language="SOL25">
    <class name="Main" parent="Object">
        <method selector="run">
            <block arity="0">
                <assign order="1">
                    <var name="x"/>
                    <expr>
                        <send selector ="new">
                            <expr>
                                <literal class="class" value="Integer"/>
                            </expr>
                        </send>
                    </expr>
                </assign>
            </block>
        </method>
    </class>
</program>